### DISCUSSION TITLE
Real time validation for multiple files

### DISCUSSION THREAD

**Discussion by Bakusedu** [livewire/livewire#2764] [1. Help]
Does livewire support real time validation for **Multiple Files** just as it supports for **Single Files**
something of this sort
`
public $photos = [];
public function updatedPhotos()
  {
        $this->validate([
            'photos.*' => 'image|max:1024', // 1MB Max
        ]);
  }
`
Thanks in anticipation

**Comment by kevin4dhd** [comment]
any solution?

**Comment by zdarin** [comment]
Im searching for this info too

**Comment by UmeshKrRana** [comment]
You can implement real-time file validation for multiple file uploads using this way in Livewire 2
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;
use Livewire\WithFileUploads;
use App\Models\File;

class FileUpload extends Component
{
    // use livewire trait
    use WithFileUploads;

    public $files = [];

    public function render()
    {
        return view('livewire.file-upload');
    }

    protected $rules = [
        'files' => 'required',
        'files.*' => 'mimes:pdf,csv,xls,xlsx,doc,docx|max:2048'
    ];

    protected $messages = [
        'files.required' => 'Please upload a file',
        'files.*.mimes' => 'Please upload a valid file (Allowed file pdf, csv, xls, xlsx, doc, docx',
        'files.*.max' => 'The uploaded file size is too large. Max allowed size 2MB',
    ];

    public function updatedFiles() {
        $this->validate();
    }

    public function submitForm() {
        $this->validate();
        dd($this->files);
    }
}

```

**Comment by tonyuh** [comment]
I'm still having this issue, the real-time validation only works on single file. For single file it validates the input file as soon as it is selected but for multiple files, for some reason it only validates after I click the form submit. I used the example code here https://livewire.laravel.com/docs/uploads#handling-multiple-files. I'm still on Laravel 10 and Livewire 3, is that an issue?

**Reply by nuocgansoi** [reply]
The same with me.
